<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DaDuDa</class>
 <widget class="QMainWindow" name="DaDuDa">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>540</width>
    <height>560</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>540</width>
    <height>560</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>540</width>
    <height>560</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>DaDuDa</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>351</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>Programmer (-c)</string>
    </property>
    <widget class="QComboBox" name="programmerCB">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>331</width>
       <height>24</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>usbasp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>usbasp-clone</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>usbtiny</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>arduino</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>arduino-ft232r</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>arduinoisp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>arduinoisporg</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>stk500</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>stk500hvsp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>stk500pp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>stk500v1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>stk500v2</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>stk600</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>stk600hvsp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>stk600pp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>atmelice_isp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>atmelice_pdi</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>atmelice_updi</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>avr109</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>avr910</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>avr911</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>avrftdi</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>avrisp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>avrisp2</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>avrispmkII</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>avrispv2</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>butterfly</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>butterfly_mk</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>flip1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>flip2</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ft232h</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ft232r</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ft245r</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>iseavrprog</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag1slow</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag2</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag2avr32</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag2dw</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag2fast</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag2isp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag2pdi</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag2updi</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag2fast</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag3</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag3dw</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag3isp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag3pdi</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtag3updi</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtagkey</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtagmkI</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtagmkII</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>jtagmkII_</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>dragon_dw</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>dragon_hvsp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>dragon_isp</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>dragon_jtag</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>dragon_pdi</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>dragon_pp</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>18</x>
       <y>60</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Port (-P)</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>148</x>
       <y>60</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Baud (-b)</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>60</y>
       <width>91</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>SCK rate (-B)</string>
     </property>
    </widget>
    <widget class="QComboBox" name="portCB">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>81</width>
       <height>24</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>usb</string>
      </property>
     </item>
    </widget>
    <widget class="QLineEdit" name="baudTB">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>80</y>
       <width>101</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="sckCB">
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>80</y>
       <width>101</width>
       <height>24</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>375 KHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1.5 MHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>750 KHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>375 KHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>187.5 KHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>93.75 KHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>32 KHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>16 KHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>8 KHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>4 KHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2 KHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>1 KHz</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>500 Hz</string>
      </property>
     </item>
    </widget>
    <widget class="QPushButton" name="scanBTN">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>80</y>
       <width>21</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>P</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>351</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Flash</string>
    </property>
    <widget class="QPushButton" name="openBTN">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Open</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="fPathTB">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>30</y>
       <width>241</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="fReadBTN">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>60</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Read</string>
     </property>
    </widget>
    <widget class="QPushButton" name="fWriteBTN">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>60</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Write</string>
     </property>
    </widget>
    <widget class="QPushButton" name="fVerifyBTN">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>60</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Verify</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>230</y>
      <width>351</width>
      <height>91</height>
     </rect>
    </property>
    <property name="title">
     <string>Eeprom</string>
    </property>
    <widget class="QPushButton" name="eeOpenBTN">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Open</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="eePathTB">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>30</y>
       <width>241</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="eeReadBTN">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>60</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Read</string>
     </property>
    </widget>
    <widget class="QPushButton" name="eeWriteBTN">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>60</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Write</string>
     </property>
    </widget>
    <widget class="QPushButton" name="eeVerifyBTN">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>60</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Verify</string>
     </property>
    </widget>
   </widget>
   <widget class="QTextEdit" name="consoleTB">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>360</y>
      <width>521</width>
      <height>151</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="maximumSize">
     <size>
      <width>521</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <property name="lineWrapMode">
     <enum>QTextEdit::NoWrap</enum>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>10</y>
      <width>161</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>Device (-p)</string>
    </property>
    <widget class="QComboBox" name="deviceCB">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>141</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="checkBTN">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>60</y>
       <width>80</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Check</string>
     </property>
    </widget>
    <widget class="QLabel" name="signLB">
     <property name="geometry">
      <rect>
       <x>78</x>
       <y>90</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>90</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Sign:</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>130</y>
      <width>161</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>Fuse and Lock bits</string>
    </property>
    <widget class="QLineEdit" name="hFuseTB">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>30</y>
       <width>31</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="lFuseTB">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>60</y>
       <width>31</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="eFuseTB">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>90</y>
       <width>31</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="lockTB">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>140</y>
       <width>31</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>FF</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Hi</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Low</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Ext</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Lock</string>
     </property>
    </widget>
    <widget class="QPushButton" name="rFuseBTN">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>30</y>
       <width>61</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Read</string>
     </property>
    </widget>
    <widget class="QPushButton" name="wFuseBTN">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>60</y>
       <width>61</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Write</string>
     </property>
    </widget>
    <widget class="QPushButton" name="wLockBTN">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>140</y>
       <width>31</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>W</string>
     </property>
    </widget>
    <widget class="QPushButton" name="rLockBTN">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>140</y>
       <width>31</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>R</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="eraseBTN">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>320</y>
      <width>80</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Erase</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="disErrCHB">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>330</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>No erase(-D)</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="noCheckSignCHB">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>330</y>
      <width>121</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>No ch sign(-F)</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="noVerifyCHB">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>330</y>
      <width>111</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>No verify(-V)</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>540</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
